Mathlib/Algebra/Group/WithOne/Defs.lean

```diff
@@ -367,18 +368,26 @@ instance addMonoidWithOne [AddMonoidWithOne α] : AddMonoidWithOne (WithZero α)
           rw [Nat.cast_succ, coe_add, coe_one]
       }
 
+instance leftDistribClass [AddZeroClass α] [MulZeroClass α] [LeftDistribClass α]  :
+    LeftDistribClass (WithZero α) where
+  left_distrib a b c := by
+    cases' a with a; · rfl
+    cases' b with b <;> cases' c with c <;> try rfl
+    exact congr_arg some (left_distrib _ _ _)
+
+instance rightDistribClass [AddZeroClass α] [MulZeroClass α] [RightDistribClass α]  :
+    RightDistribClass (WithZero α) where
+  right_distrib a b c := by
+    cases' c with c
+    · change (a + b) * 0 = a * 0 + b * 0
+      simp
+    cases' a with a <;> cases' b with b <;> try rfl
+    exact congr_arg some (right_distrib _ _ _)
+
```

Comment:
Can you add an instance of `Distrib` too for good measure?

Reply:
Ok I have added the `Distrib` instance but I had to duplicate all the code, I don't really get why I can't get a `Distrib` from a `leftDistrib` and a `rightDistrib` by typeclass inference.

Reply:
Can you put the instances in the other order? You should be able to at least use `mul_add` and `add_mul` to prove `Distrib`

Reply:
Lean is telling me that the `Mul` of the `Distrib` is not the same as the `Mul` of the `LeftDistribClass`

Reply:
I think that was due to your unused variables you just deleted

Reply:
This all seems to have arisen from using `[AddZeroClass α] [MulZeroClass α]` when `[Mul α] [Add α]` sufficed